osu.Game/Screens/Select/DifficultyRecommender.cs

```diff
@@ -75,12 +89,35 @@ private void calculateRecommendedDifficulties()
             });
         }
 
+        private IEnumerable<RulesetInfo> getBestRulesetOrder()
+        {
+            IEnumerable<RulesetInfo> bestRulesetOrder = recommendedStarDifficulty.OrderByDescending(pair => pair.Value)
+                                                                                 .Select(pair => pair.Key)
+                                                                                 .ToList();
+
+            List<RulesetInfo> orderedRulesets;
+
+            if (bestRulesetOrder.Contains(ruleset.Value))
+            {
+                orderedRulesets = bestRulesetOrder.ToList();
+                orderedRulesets.Remove(ruleset.Value);
+                orderedRulesets.Insert(0, ruleset.Value);
+            }
+            else
+            {
+                orderedRulesets = new List<RulesetInfo> { ruleset.Value };
+                orderedRulesets.AddRange(bestRulesetOrder);
+            }
+
+            return orderedRulesets;
+        }
+
         private void onlineStateChanged(ValueChangedEvent<APIState> state) => Schedule(() =>
         {
             switch (state.NewValue)
             {
                 case APIState.Online:
-                    calculateRecommendedDifficulties();
+                    calculateRecommendedDifficulties(true);
```

Comment:
probably want to clear any values on going offline

Reply:
Not sure there are any to clear. After we have been online once we never need to change anything again.

Reply:
What if the user logs in as a different user?

Reply:
User IDs are now checked, but still no action needs to be taken if the user logs out, unless we don't want to continue recommending the last logged in user's star rating. I think if somebody wants to play offline, they would still expect to get correct recommendations.

Reply:
Honestly it's debatable that it shouldn't be cleared. I can see an argument for clearing, too (say, having someone new play). We obviously won't know for sure which one it is, though, so I'm not sure I care much either way. This entire feature is almost too subtle to notice the existence of anyway.

Reply:
Also wanted to note, lest I multi account, I don't think I can test this.

> too subtle to notice the existence of anyway

In song select it is very noticeable. in presentBeatmap's case, I can agree.

Reply:
I've never consciously been irked by (or even noted) the extra two or three arrow presses that it takes to select a difficulty in song select I want. Different strokes for different folks I suppose.